src/client.ts

```diff
@@ -4124,10 +4368,13 @@ export class MatrixClient extends EventEmitter {
      * @param {string} info The kind of info to set (e.g. 'avatar_url')
      * @param {Object} data The JSON object to set.
      * @param {module:client.callback} callback Optional.
-     * @return {Promise} Resolves: TODO
+     * @return {Promise} Resolves: to an empty object {}
      * @return {module:http-api.MatrixError} Rejects: with an error response.
      */
-    public setProfileInfo(info: string, data: any, callback?: Callback): Promise<void> {
+    // eslint-disable-next-line camelcase
+    public setProfileInfo(info: "avatar_url", data: { avatar_url: string }, callback?: Callback): Promise<{}>;
+    public setProfileInfo(info: "displayname", data: { displayname: string }, callback?: Callback): Promise<{}>;
+    public setProfileInfo(info: string, data: object, callback?: Callback): Promise<{}> {
```

Comment:
instead of `object` we should use an `IProfile` type with optional `displayname` and `avatar_url`. This would also alleviate the need for overloads.

Reply:
but the overloads are stricter, they say that depending on `info` you can only provide one field or the other, like the spec says.

Reply:
ah, indeed. Fair enough.